libs/iavl/nodedb.go

```diff
@@ -638,6 +640,48 @@ func (ndb *nodeDB) deleteOrphans(batch dbm.Batch, version int64) {
 	})
 }
 
+func (ndb *nodeDB) deleteOrphansFromDB(version int64) {
+	// Will be zero if there is no previous version.
+	predecessor := ndb.getPreviousVersion(version)
+
+	// Traverse orphans with a lifetime ending at the version specified.
+	// TODO optimize.
+	ndb.traverseOrphansVersion(version, func(key, hash []byte) {
+		var fromVersion, toVersion int64
+
+		// See comment on `orphanKeyFmt`. Note that here, `version` and
+		// `toVersion` are always equal.
+		orphanKeyFormat.Scan(key, &toVersion, &fromVersion)
+
+		// Delete orphan key and reverse-lookup key.
+		err := ndb.db.Delete(key)
```

Comment:
Should we use batch here? I think a Orphans key map a node key. If the Orphans key lost. then The node key will stay in db forever? Maybe we can split the big batch to small ones.